packages/wast-parser/src/fsm.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+const STOP = Symbol("STOP");
+
+type State<T> = T | Symbol;
+type TransitionEdge<Q> = () => [Q, number] | false;
+type TransitionFunction<Q> = () => [Q, number];
+type TransitionList<Q> = { [Q]: Array<TransitionEdge<Q>> };
+
+function makeTransition<Q>(
+  regex: RegExp,
+  nextState: Q,
+  n: number = 1,
+  allowedSeparator: ?string
```

Comment:
The API would be clearer to me if you passed an object here for `n` and `allowedSeparator`.

Its usage would then be:
```js
makeTransition(/[0-9]/, "DEC_FRAC", { n: 1, allowedSeparator: NUMERIC_SEPARATOR }),
```

Reply:
Very good point, this way I don't need to pass `n: 1` for every separator, thanks!